content/help/troubleshooting/index.md

```diff
@@ -8,6 +8,120 @@ draft: true
 
 Oh no! You're having trouble? Below is a list of solutions to common problems.
 
+## 503 errors while reconfiguring service routes
+
+When setting route rules to direct traffic to specific versions (subsets) of a service, care must be taken to ensure
+that the subsets are available before they are used in the routes. Otherwise, calls to the service may return
+503 errors during a reconfiguration period.
+
+Creating both the `VirtualServices` and `DestinationRules` that define the corresponding subsets using a single `istioctl`
+call (e.g., `istioctl create -f myVirtualServiceAndDestinationRule.yaml` is not good enough because the
+propagation of resources from Pilot to the Envoys is asynchronous, so there is no guarantee that the
+subsets will be available before they are needed, possibly resulting in a small window of time where 503 errors can occur.
+
+To make sure services will have zero down-time when configuring routes, configure rules as follows:
+
+* Update `DestinationRules` to add a new subset first, before updating any `VirtualServices` that use it.
+
+    1. `istioctl create -f destinationRuleWithSubsetAbc.yaml`
+
+    1. Wait for the `DestinationRule` configuration to propagate to the Envoys
+
+    1. `istioctl create -f virtualServiceUsingSubsetAbc.yaml`
+
+* Update `VirtualServices` to remove any references to a subset, before removing the subset from a `DestinationRule`.
+
+    1. `istioctl replace -f virtualServicePreviouslyUsingSubsetAbc.yaml`
+
+    1. Wait for the `VirtualService` configuration to propagate to the Envoys
+
+    1. `istioctl replace -f destinationRuleWithRemovedSubsetAbc.yaml`
+
+## Route rules have no effect on ingress gateway requests
+
+Let's assume you are using an ingress `Gateway` and corresponding `VirtualSerive` to access an internal service.
+For example, your `VirtualService` looks something like this:
+
+```yaml
```

Comment:
Please update this PR to use the new annotations for code blocks. Instead of backticks, you need to use

{{< text bash >}}

Please see [here](https://preliminary.istio.io/about/contribute/writing-a-new-topic/#embedding-preformatted-blocks) for info

Thanks.